radis/io/hitran.py

```diff
@@ -340,6 +340,239 @@ def hit2df(
     return df
 
 
+def hit2df_hapi(
+    fname,
+    cache=True,
+    verbose=True,
+    drop_non_numeric=True,
+    load_wavenum_min=None,
+    load_wavenum_max=None,
+    engine="pytables",
+    parse_quanta=True,
+    data_file=None,
+):
+    """Convert a HITRAN/HITEMP [1]_ file to a Pandas dataframe
+
+    Parameters
+    ----------
+    fname: str
+        HITRAN-HITEMP file name
+    cache: boolean, or ``'regen'`` or ``'force'``
+        if ``True``, a pandas-readable HDF5 file is generated on first access,
+        and later used. This saves on the datatype cast and conversion and
+        improves performances a lot (but changes in the database are not
+        taken into account). If False, no database is used. If ``'regen'``, temp
+        file are reconstructed. Default ``True``.
+
+    Other Parameters
+    ----------------
+    drop_non_numeric: boolean
+        if ``True``, non numeric columns are dropped. This improves performances,
+        but make sure all the columns you need are converted to numeric formats
+        before hand. Default ``True``. Note that if a cache file is loaded it
+        will be left untouched.
+    load_wavenum_min, load_wavenum_max: float
+        if not ``'None'``, only load the cached file if it contains data for
+        wavenumbers above/below the specified value. See :py:func`~radis.io.cache_files.load_h5_cache_file`.
+        Default ``'None'``.
+    engine: 'pytables', 'vaex'
+        format for Hdf5 cache file. Default `pytables`
+    parse_quanta: bool
+        if ``True``, parse local & global quanta (required to identify lines
+        for non-LTE calculations ; but sometimes lines are not labelled.)
+
+    Returns
+    -------
+    df: pandas Dataframe
+        dataframe containing all lines and parameters
+
+
+    References
+    ----------
+
+    .. [1] `HITRAN 1996, Rothman et al., 1998 <https://www.sciencedirect.com/science/article/pii/S0022407398000788>`__
+
+
+
+    Notes
+    -----
+
+    Performances: see CDSD-HITEMP parser
+
+
+    See Also
+    --------
+
+    :func:`~radis.io.cdsd.cdsd2df`
+    """
+    from hapi import LOCAL_TABLE_CACHE
+
+    data_file_name = data_file.split(".")
+    df = pd.DataFrame(LOCAL_TABLE_CACHE[data_file_name[0]]["data"])
```

Comment:
isn't `data_file`   `fname` here ? 

If it's different add docstrings for data_file

Also, the line `data_file_name = data_file.split(".")` acn be better handled by os.path or pathlib.

Reply:
Yes, actually I can also extract the same data from` fname `as well, so I have removed the `data_file` in the recent commit